homeassistant/components/binary_sensor/doorbird.py

```diff
@@ -1,32 +1,52 @@
 """Support for reading binary states from a DoorBird video doorbell."""
-from datetime import timedelta
+import asyncio
+import datetime
 import logging
 
 from homeassistant.components.binary_sensor import BinarySensorDevice
 from homeassistant.components.doorbird import DOMAIN as DOORBIRD_DOMAIN
-from homeassistant.util import Throttle
+from homeassistant.components.http import HomeAssistantView
 
 DEPENDENCIES = ['doorbird']
 
 _LOGGER = logging.getLogger(__name__)
-_MIN_UPDATE_INTERVAL = timedelta(milliseconds=250)
+
+API_URL = "/api/" + DOORBIRD_DOMAIN
+
+SENSOR_DOORBELL = "doorbell"
 
 SENSOR_TYPES = {
-    "doorbell": {
+    SENSOR_DOORBELL: {
         "name": "Doorbell Ringing",
         "icon": {
             True: "bell-ring",
             False: "bell",
             None: "bell-outline"
-        }
+        },
+        "time": datetime.timedelta(seconds=5),
+        "instance": None
     }
 }
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the DoorBird binary sensor component."""
     device = hass.data.get(DOORBIRD_DOMAIN)
-    add_devices([DoorBirdBinarySensor(device, "doorbell")], True)
+
+    # Provide an endpoint for the device to call to trigger events
+    hass.http.register_view(DoorbirdRequestView())
+    _LOGGER.debug("Registered DoorBird request view")
+
+    # This will make HA the only service that gets doorbell events
```

Comment:
This should be made clear in the docs.

Reply:
https://github.com/Klikini/home-assistant.github.io/commit/a97ec23d2a8757bea9be2e698a14d55fe5d5d8e7